stdlib/domain.ml

```diff
@@ -137,15 +137,11 @@ let spawn f =
   let termination_mutex = Mutex.create () in
   let state = Atomic.make Running in
   let body () =
-    DLS.create_dls ();
-    let res = f () in
-    do_at_exit ();
-    res
-  in
-  let wrapped_body () =
-    let result = match body () with
+    let result = match DLS.create_dls (); f () with
       | x -> Ok x
-      | exception ex -> Error ex in
+      | exception ex -> Error ex
+    in
+    do_at_exit ();
```

Comment:
Given that `do_at_exit ()` may also throw an exception (the API exposes this to user defined functions), we probably need to guard it as well. Otherwise an at_exit handler which throws in a child domain is going to hang domains that try to join it.

Reply:
I've changed the behaviour in https://github.com/ocaml-multicore/ocaml-multicore/pull/586/commits/4ce92d14376ffca1ed2580f5b0150a61d7059891 such that exceptions from `Domain.at_exit` callbacks are ignored. The documentation also mentions this. Sadly, this is not the case with `Stdlib.at_exit`, which can prevent other `at_exit` functions from running since the functions are evaluated in LIFO order. 

@talex5 @craigfe Wondered if you had an opinion on this?

Reply:
I never use `at_exit`, so I don't really mind, though I'm generally against anything that silently drops exceptions.